### PR TITLE
Small changes from Yocto/musl builds

### PR DESCRIPTION
**Pull request by AdrianBunk**
These are 4 small changes that would look correct even if they wouldn't be needed for fixing musl builds, and except for the ULONG_LONG_MAX change they could potentially also break glibc builds after some future round of header cleanups there.

I am not sure whether all the changes and hacks currently used for getting systemd build with musl in Yocto can ever get upstreamed, this pull request is solely about upstreaming some of the "obviously correct" ones.

### CODE REVIEW

File: src/udev/udev-event.c [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <sys/wait.h>
```
Comment (yuwata) [inline src/udev/udev-event.c:10]: Please sort the headers alphabetically.

File: src/basic/socket-util.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <sys/un.h>
+#include <string.h>
```
Comment (yuwata) [inline src/basic/socket-util.h]: Same here.

File: src/udev/udev-event.c (hunk 1 above)
Comment (AdrianBunk) [reply src/udev/udev-event.c:10]: Done.

File: src/basic/socket-util.h (hunk 2 above)
Comment (AdrianBunk) [reply src/basic/socket-util.h]: Done.

**Comment by keszybz** [comment]
LGTM too.

**Pull request merged**